### PR TITLE
NcpBase: Update getter for `OFF_MESH_ROUTES` to include `NextHopIsThisDevice` field

### LINKED ISSUE #407: spinel: documentation issue with local routes property
PROP_THREAD_LOCAL_ROUTES (route add/remove) looks to have the same arguments as PROP_THREAD_ON_MESH_NETS (prefix add/remove), but the stable parameter should probably be               prf instead: Default Router Preference, which may be: 'high', 'med', or 'low'.

Also, for PROP_THREAD_ON_MESH_NETS, the stable flag should probably just be part of the general flags parameter and follow the Thread specification definition, otherwise the API has a  redundant and possibly conflicting way of accessing it.

```
D.4.12. PROP 91: PROP_THREAD_LOCAL_ROUTES

Type: Read-Write
Packed-Encoding: A(T(6CbC))
Data per item is:

6: IPv6 Prefix
C: Prefix length, in bits
b: Stable flag
C: Other flags
```

### PR DESCRIPTION
**Pull request by abtink**
The spinel header file and spinel documentation are also updated
accordingly.

Also resolves #407.

### CODE REVIEW

**Comment by abtink** [comment]
Related PR into wpantund: https://github.com/openthread/wpantund/pull/183

File: doc/spinel-protocol-src/spinel-tech-thread.md [hunk 1]
```diff
@@ -98,18 +98,22 @@ Data per item is:
...
-* `C`: Prefix length, in bits
+* `6`: Route Prefix
+* `C`: Prefix length in bits
 * `b`: Stable flag
 * `C`: Other flags
 * `b`: "Is defined locally" flag. Set if this route info was locally
   defined as part of local network data. Assumed to be true for set,
   insert and replace. Clear if the route is part of partition's network
   data.
+* `b`: "Next hop is this device" flag. Set if the next hope for the
```
Comment (jwhui) [inline doc/spinel-protocol-src/spinel-tech-thread.md]: typo: `hope` -> `hop`

File: src/ncp/spinel.h [hunk 2]
```diff
@@ -740,7 +740,28 @@ typedef enum
...
+     *
+     *  `6`: Route Prefix
+     *  `C`: Prefix length in bits
+     *  `b`: Stable flag
+     *  `C`: Other flags
+     *  `b`: "Is defined locally" flag. Set if this route info was locally
+     *       defined as part of local network data. Assumed to be true for set,
+     *       insert and replace. Clear if the route is part of partition's network
+     *       data.
+     *  `b`: "Next hop is this device" flag. Set if the next hope for the
```
Comment (jwhui) [inline src/ncp/spinel.h]: typo: `hope` -> `hop`

File: doc/spinel-protocol-src/spinel-tech-thread.md [hunk 3]
```diff
@@ -98,18 +98,22 @@ Data per item is:
...
+* Packed-Encoding: `A(t(6CbCbb))`
 
 Data per item is:
 
-* `6`: IPv6 Prefix
-* `C`: Prefix length, in bits
+* `6`: Route Prefix
+* `C`: Prefix length in bits
 * `b`: Stable flag
 * `C`: Other flags
```
Comment (jwhui) [inline doc/spinel-protocol-src/spinel-tech-thread.md]: Can we also address #407 in this PR?

File: doc/spinel-protocol-src/spinel-tech-thread.md (hunk 1 above)
Comment (abtink) [inline doc/spinel-protocol-src/spinel-tech-thread.md]: Thanks. Fixed it.

File: doc/spinel-protocol-src/spinel-tech-thread.md (hunk 3 above)
Comment (abtink) [inline doc/spinel-protocol-src/spinel-tech-thread.md]: Updated the PR to address the https://github.com/openthread/openthread/issues/407.

**Pull request merged**